onnxruntime/python/tools/quantization/qdq_quantizer.py

```diff
@@ -4,40 +4,43 @@
 # license information.
 # --------------------------------------------------------------------------
 import logging
-import os
-import struct
-from pathlib import Path
+from enum import Enum
 
-import numpy as np
 import onnx
 import onnx.numpy_helper
 from onnx import TensorProto
 from onnx import onnx_pb as onnx_proto
 
-from onnxruntime import GraphOptimizationLevel, InferenceSession, SessionOptions
-
-from .onnx_model import ONNXModel
 from .onnx_quantizer import ONNXQuantizer
 from .quant_utils import (
-    QuantizationMode,
-    QuantizedInitializer,
+    DEQUANT_OP_NAME,
+    QUANT_OP_NAME,
     QuantizedValue,
     QuantizedValueType,
-    QuantType,
     __producer__,
     __version__,
-    attribute_to_kwarg,
+    add_dequant_output_suffix,
+    add_dequant_suffix,
+    add_quant_input_suffix,
+    add_quant_output_suffix,
+    add_quant_suffix,
     find_by_name,
-    generate_identified_filename,
-    get_elem_index,
-    get_mul_node,
-    onnx_domain,
-    quantize_nparray,
-    type_to_name,
 )
 from .registry import CreateQDQQuantizer
 
 
+class QDQQuantTensorType(Enum):
+    NORMAL = 0
+    SHARE_PARAM = 1
+
+
+class QDQTensorQuantInfo:
+    def __init__(self, tensor_type=QDQQuantTensorType.NORMAL, quant_para_provider=None, axis=None):
+        self.tensor_type = tensor_type
+        self.quant_para_provider = quant_para_provider
+        self.axis = axis
+
+
 class QDQQuantizer(ONNXQuantizer):
     def __init__(
```

Comment:
not part of this pr, but: W0102: Dangerous default value {} as argument (dangerous-default-value). Use None instead